### ISSUE TITLE
Error loading library "x32gui.dll"

### ISSUE THREAD

**Issue by ax330d** [x64dbg/x64dbg#866]
This error happens on latest Windows 10 x64, version of x64dbg snapshot_2016-07-17_11-31, when trying to open x32dbg.exe.

**Comment by torusrxxx** [comment]
Build error at DisassemblerGraphView.cpp:1208 : ambigous template.

**Comment by mrexodia** [comment]
Strange... It compiled fine for me.

**Comment by mrexodia** [comment]
I assume this is fixed now?

**Comment by ax330d** [comment]
Yes, it is fixed.

**Comment by kjaisinghyadav** [comment]
still i am getting this  error, when i run the open source code in visual studio 2017

**Comment by mrexodia** [comment]
You have to compile the GUI first...

**Comment by rh101** [comment]
I had the exact same issue, and it had nothing to do with the GUI DLL.  The steps in one of the videos showing how to compile x64dbg in VS2017 is incomplete for the current release.  You need to make sure ldconvert.dll is also copied into the bin output folder, otherwise you will get the 'Error loading library "x32gui.dll"' message.  

That ldconvert.dll file wasn't included in the instructions in this video: https://vimeo.com/213004417

**Comment by mrexodia** [comment]
You can run setupdeps.bat to copy all dependencies

**Comment by rh101** [comment]
@mrexodia Yes, that is the case, but following the steps in the video linked in the wiki regarding compiling with VS2017/QT 5.8 (https://github.com/x64dbg/x64dbg/wiki/Compiling-the-whole-project#video), there is no mention of the setupdeps.bat.  If someone is attempting this for the first time then you can imagine the frustration they would have when they come across this issue, especially since the error message is misleading (it mentions x32gui.dll when it really should have mentioned the actual missing DLL).  It's just that the video is a bit dated, that's all, and perhaps mentioning something regarding setupdeps.bat in that wiki page may help.  I'll see if I have access to update the wiki entry.

**Comment by mrexodia** [comment]
The wiki clearly states that vs2017 is not supported, but feel free to
update the entry :)

**Comment by rh101** [comment]
@mrexodia Done :)

**Comment by mrexodia** [comment]
:+1:

**Comment by openGLnoob** [comment]
Got this error on my old XP x86 SP3 with the last snapshot_2022-10-28_15-42. If it started with compatible mode I get the error, if not - this process close immediately. How can I run x32dbg with XP ,does it need to be compiled?

**Comment by mrexodia** [comment]
It’s not working on XP currently. You can use an older snapshot.

**Comment by openGLnoob** [comment]
Mrexodia, which is the last version of snapshots supports XP SP3? I download it randomly, 67-8 versions, but only snapshots from 2014 works with XP for me. Thank you!

**Comment by mrexodia** [comment]
Snapshots from a few months ago are working. But for XP support you’re on
your own. Windows 7 shouldn’t be an issue

**Comment by Qwerterwer** [comment]
Ok... I've tried about 20 snapshots from 2014 to now and only 2014 snapshots works "out of the box" on XP. All this works fine on Win7.
I tried compiling the latest source in according with the wiki. I installed VS2013, Qt5.6.3 x86 for msvc2013 and used both version of qtcreator (native and 4.3.1). Everything builds without problem. I tried to run on XP with your deps dll's and with msvc_2013 dll's but every time I get "Error loading library "x32gui.dll"". While on Win 7 the everything starts without problem with different dll's. When I debug the loading process all Qt dll's and x32gui.dll are loaded successfully, but I get the error in Qt5Core.dll with "C000001D, EXCEPTION_ILLEGAL_INSTRUCTION" on 67011330 RVA. I'll keep trying....
